### PR TITLE
make `jj log --patch --summary` show both summary and diff

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: src/diff_util.rs [hunk 1]
```diff
@@ -54,20 +52,37 @@ pub enum DiffFormat {
...
 }
 
-pub fn diff_format_for(ui: &Ui, args: &DiffFormatArgs) -> DiffFormat {
-    if args.summary {
-        DiffFormat::Summary
-    } else if args.git {
-        DiffFormat::Git
-    } else if args.color_words {
-        DiffFormat::ColorWords
+/// Returns a list of requested diff formats, which never be empty.
```
Comment (martinvonz) [inline src/diff_util.rs:55]: nit: missing word (probably "will")

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### New features
 
+* `jj log --summary --patch` now shows both summary and diff outputs.
```
Comment (martinvonz) [inline CHANGELOG.md:14]: IIUC, we also allow e.g. `jj diff -s --git` now. I'm not sure that's useful enough to be worth mentioning here. But maybe we should have a test for one of those cases?

**Review by martinvonz — APPROVED** [review]
Interesting feature. I had not considered this myself. I didn't know that `hg log -p --stat` was allowed.

File: CHANGELOG.md (hunk 2 above)
Comment (yuja) [reply CHANGELOG.md:17]: Good point. Added test for that.

Comment (martinvonz) [reply CHANGELOG.md:17]: Looks like you added one for `log`. Did you mean for that to be for `diff`? I don't mind having one for `log`, but we also support `jj diff -s --git`, right?

File: src/diff_util.rs [hunk 3]
```diff
@@ -54,20 +52,37 @@ pub enum DiffFormat {
...
+}
+
+fn diff_formats_from_args(args: &DiffFormatArgs) -> Vec<DiffFormat> {
+    [
+        (args.summary, DiffFormat::Summary),
+        (args.git, DiffFormat::Git),
+        (args.color_words, DiffFormat::ColorWords),
+    ]
+    .into_iter()
+    .filter_map(|(p, f)| p.then(|| f))
```
[the remark this answers is not in the document]
Comment (yuja) [reply src/diff_util.rs]: Yep, that's why. I'm not supper happy with bumping MSRV for small language features, but I'm not against either. Anyway, we'll probably want to use let-else and GAT at some point.

[the remark this answers is not in the document]
Comment (yuja) [reply src/diff_util.rs]: Changed to `(arg, format)`

File: CHANGELOG.md (hunk 2 above)
Comment (yuja) [reply CHANGELOG.md:17]: Oops, fixed.

Comment (martinvonz) [reply CHANGELOG.md:17]: Looks good now. Thanks for adding both tests :)

**Pull request merged**